src/api/app/lib/routes_helper/role_matcher.rb

```diff
@@ -0,0 +1,14 @@
+module RoutesHelper
+  class RoleMatcher
+    def self.matches?(request)
+      return false if request.bot?
+
+      # TODO: This doesn't work when proxy is enabled, so use code from Webui::WebuiController#check_user
+      current_user_login = request.session[:login]
+      current_user = current_user_login.present? ? User.find_by_login(current_user_login) : User.possibly_nobody
+
+      binding.pry
```

Comment:
:eyes:

Reply:
Yep... I still need to test some things hehehe :slightly_smiling_face: